hpke/hpke.go

```diff
@@ -67,20 +67,19 @@ const (
 
 // Suite is an HPKE cipher suite consisting of a KEM, KDF, and AEAD algorithm.
 type Suite struct {
-	KemID  KemID
-	KdfID  KdfID
-	AeadID AeadID
-	_      struct{}
+	kemID  KEM
+	kdfID  KDF
+	aeadID AEAD
 }
 
-// NewSuite builds a Suite from a specified set of algorithms. Panics if an
-// algorithm identifier is not valid.
-func NewSuite(kemID KemID, kdfID KdfID, aeadID AeadID) Suite {
-	s := Suite{KemID: kemID, KdfID: kdfID, AeadID: aeadID}
+// NewSuite builds a Suite from a specified set of algorithms. Returns error
+// if an algorithm identifier is not valid.
+func NewSuite(kemID KEM, kdfID KDF, aeadID AEAD) (*Suite, error) {
```

Comment:
I'd rather panic then propagate an error here. The user can check if a suite is valid via `kemID.IsValid() && kdfID.IsValid() && aeadID.isValid()`.